gateway/models.py

```diff
@@ -131,33 +181,22 @@ def __init__(
         self.pairing_pass_key = pairing_pass_key
         self.pairing_oob_key = pairing_oob_key
         self.pairing_oobrn = pairing_oobrn
-        if endpoint_apps:
-            self.endpoint_apps.extend(endpoint_apps)
-        self.created_time = created_time
-        self.modified_time = created_time
 
     def __repr__(self):
-        return f"<id {self.id}>"
+        return f"<id {self.device_id}>"
 
     def serialize(self):
         """serialize function"""
         response = {
-            "schemas": ["urn:ietf:params:scim:schemas:core:2.0:Device",
-                        "urn:ietf:params:scim:schemas:extension:ble:2.0:Device"],
-            "id": self.id,
-            "deviceDisplayName": self.device_display_name,
-            "adminState": self.admin_state,
             "urn:ietf:params:scim:schemas:extension:ble:2.0:Device": {
                 "versionSupport": self.version_support,
-                "deviceMacAddress": self.device_mac_address,
                 "isRandom": self.is_random,
                 "pairingMethods": self.pairing_methods,
-            },
-            "meta": {"resourceType": "Device",
-                     "created": self.created_time,
-                     "lastModified": self.modified_time},
+            }
         }
 
+        if self.device_mac_address:
+            response['deviceMacAddress'] = self.device_mac_address
```

Comment:
Actually, this should be
```suggestion
            response["urn:ietf:params:scim:schemas:extension:ble:2.0:Device"]["deviceMacAddress"] = self.device_mac_address
```
Or it is probably easier to add it above as such: 
```python
        response = {
            "urn:ietf:params:scim:schemas:extension:ble:2.0:Device": {
                "versionSupport": self.version_support,
                "deviceMacAddress": self.device_mac_address,
                "isRandom": self.is_random,
                "pairingMethods": self.pairing_methods,
            }
        }
```